src/contracts/GPv2Settlement.sol

```diff
@@ -0,0 +1,106 @@
+    ///   spread. Specifically the price is either:
+    ///   - Effective price in case the AMM was traded with,
+    ///   - Spot price in case the AMM was not traded with.
+    /// - All orders are valid and signed
+    /// - Accounts have sufficient balance and approval.
+    ///
+    /// Note that orders in this method are expressed as encoded bytes. These
+    /// bytes are tightly packed orders encoded with `abi.encodePacked` in order
+    /// to reduce the amount of call data required to call this method. Orders
+    /// encode the following fields:
+    /// ```
+    /// struct Order {
+    ///     sellAmount:     uint112,
+    ///     buyAmount:      uint112,
+    ///     executedAmount: uint112,
+    ///     tip:            uint112,
+    ///     nonce:          uint32,
+    ///     validTo:        uint32,
+    ///     flags:          uint8,
+    ///     signature: {
+    ///         v:          uint8,
+    ///         r:          bytes32,
+    ///         s:          bytes32,
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// Note that the encoded order data does not contain which token is the
+    /// sell or buy token. This data is implicit from which parameter the order
+    /// was specified from.
+    ///
+    /// @param token0 The address of token 0 being traded in the batch.
+    /// @param token1 The address of token 1 being traded in the batch.
+    /// @param d0 The amount of token 0 being traded to the AMM.
+    /// @param d1 The amount of token 1 being traded to the AMM.
+    /// @param clearingPrice0 The price of token 0 expressed in arbitrary units.
+    /// Note the exchange rate between token 0 and token 1 is `price1 / price0`.
+    /// @param clearingPrice1 The price of token 1 expressed in arbitrary units.
+    /// Note the exchange rate between token 1 and token 0 is `price0 / price1`.
+    /// @param encodedOrders0 All orders trading token 0 for token 1, that is,
+    /// either orders selling token 0 for token 1 or buying token 1 for token 0.
+    /// @param encodedOrders1 All orders trading token 1 for token 2.
+    function settle(
+        IERC20 token0,
+        IERC20 token1,
+        int256 d0,
+        int256 d1,
+        uint256 clearingPrice0,
+        uint256 clearingPrice1,
+        bytes calldata encodedOrders0,
+        bytes calldata encodedOrders1
+    ) external {
+        revert("not yet implemented");
+    }
+
+    /// @dev Returns a unique ID for the specified token pair. Note that the
+    /// order in which the tokens are specified does not matter.
+    /// @param token0 The address one of the tokens of the pair.
+    /// @param token1 The address the other token of the pair.
+    /// @return The token ID unique to the pair.
+    function pairId(IERC20 token0, IERC20 token1)
+        public
+        pure
+        returns (bytes20)
+    {
+        require(token0 != token1, "invalid pair");
+        return (bytes20(address(token0)) ^ bytes20(address(token1)));
```

Comment:
We should keep in mind that this is not collision resistant when we use this function.

Reply:
Is this a bad idea then? Should we do what Uniswap does and require tokens to be in lexographical order?

Reply:
It depends on how it's used in the end, but I'd say it's better to copy Uniswap, so we can use the pair id to reliably identify a single pair.

Reply:
I think we can use the Uniswap token pair address for the mapping then.

Reply:
I'll leave it as is for now, and open a follow up PR with this idea.

Reply:
Opened #84